jms/src/main/scala/akka/stream/alpakka/jms/JmsProducerStage.scala

```diff
@@ -155,8 +206,7 @@ private[jms] object JmsProducerStage {
    *
    * To get a condensed view of what the Holder is about, have a look there too.
    */
-  class Holder[A](var elem: Try[A], val cb: AsyncCallback[Holder[A]], val jmsProducer: Option[JmsMessageProducer])
-      extends (Try[A] => Unit) {
+  class Holder[A](var elem: Try[A]) extends (Try[A] => Unit) {
```

Comment:
future callback is not needed, producer is handled outside of the message, so the holder got a lot simpler.